### PR TITLE
Add support for Wayland wl-clipboard on Linux

### PR DESCRIPTION
**Pull request by nacnudus**
clipr currently supports X11 on Linux. Wayland is a replacement for X11 that is gathering support.  The [`wl-clipboard`](https://github.com/bugaevc/wl-clipboard) tool provides a clipboard for Wayland via the commans `wl-copy` and `wl-paste`.  This commit adds support for those commands.

### CODE REVIEW

**Comment by nacnudus** [comment]
Not sure the Travis error is related to this PR:

```
Installing package dependencies
0.01s$ sh -e /etc/init.d/xvfb start
sh: 0: Can't open /etc/init.d/xvfb
The command "sh -e /etc/init.d/xvfb start" failed and exited with 127 during .
```

**Comment by mdlincoln** [comment]
@nacnudus thanks very much for submitting this. I'm always happy to see ways to extend clipr to other clipboard implementations.

`sh -e /etc/init.d/xvfb start` is part of the init script that we've usually used to simulate a screen on headless travis. They've obviously changed something in their build containers for how this works 🤦‍♂

While this problem isn't directly related to your PR, it would be _ideal_ if we could get the _travis.yml configuration set up so at least one of the builds runs on a container that only has `wl-copy` so that we can actually test this case. I'm not sure if that will be possible without investigating Travis' current configurations more, so I'll take some time to look at that - suggestions heartily welcomed!

File: R/linux_clipboard.R [hunk 1]
```diff
@@ -27,6 +27,12 @@ has_xclip <- function() has_util(c("xclip", "-o", "-selection", "clipboard"))
 # Determine if system has 'xsel' installed
 has_xsel <- function() has_util(c("xsel", "--clipboard"))
 
+# Determine if system has 'wl-paste' installed
+has_wlpaste <- function() has_util(c("wl-paste", "--primary"))
```
Comment (mdlincoln) [inline R/linux_clipboard.R]: Please set up a test in `tests/testthat/test_systems.R` to check these functions work.

First, go to `travis.yml` and add an entry in the build matrix that checks for `wl-clipboard`. Follow the pattern that you see for `xsel` and `xclip`

Comment (mdlincoln) [reply R/linux_clipboard.R]: You'll also need to add tests ito `tests/testthat/test-diagnostics.R`, again following the pattern you'll see for `xlicp` and `xsel`

**Review by mdlincoln — CHANGES_REQUESTED** [review]
Alright, if you pull in the latest upstream commits off of the master branch, I've gotten `xvfb` working again on Travis. Let's see how wl-clipboard works now - you'll need to write up some tests so we can properly evaluate it.

File: R/linux_clipboard.R (hunk 1 above)
Comment (bugaevc) [reply R/linux_clipboard.R]: Can you please name it `has_wl_paste`? (and `has_wl_copy` below)

I don't quite like it when people spell it as `wlcopy`/`wlpaste` is all.

**Comment by nacnudus** [comment]
Thanks @mdlincoln for the clear instructions. I am currently away but I'll get to it next week.

File: R/linux_clipboard.R [hunk 2]
```diff
@@ -69,6 +81,12 @@ X11_write_clip <- function(content, object_type, breaks, eos, return_new, ...) {
     con <- pipe("xclip -i -sel p -f | xclip -i -sel c", "w")
   } else if (has_xsel()) {
     con <- pipe("xsel -b -i", "w")
+<<<<<<< HEAD
+  } else if (has_wl_copy()) {
+=======
+  } else if (has_wlcopy()) {
+>>>>>>> d790d5c26010496a70107640bb2c30464e8de1b4
+    con <- pipe("wl-copy", "w")
```
Comment (bugaevc) [inline R/linux_clipboard.R:89]: These look like merge conflicts :(

**Comment by nacnudus** [comment]
As you can tell, I am having trouble setting up Travis with wayland and wl-clipboard. Here are the [logs](https://travis-ci.org/nacnudus/clipr).

**Comment by mdlincoln** [comment]
@nacnudus thanks for your patience while I've been getting to this on my to-do list. Getting Travis to deploy how you want can be a real pain - I'm not yet entirely sure if it IS possible to get it to use wayland, but I'll take a closer look.

**Comment by mdlincoln** [comment]
I've merged in these changes as part of a larger branch of work that also sets up github actions testing. I'll be crediting all the contributors in the docs! Thank you for your help and patience 🍾

**Comment by nacnudus** [comment]
Thank you @mdlincoln and everyone who helped to get this working.  It's a huge improvement to the quality of life, working in linux.

**Pull request closed without merging**